dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBMongoDB/AzureCosmosDBMongoDBVectorStoreRecordCollection.cs

```diff
@@ -245,7 +251,8 @@ private async Task CreateIndexAsync(string collectionName, CancellationToken can
         foreach (var property in this._vectorStoreRecordDefinition.Properties.OfType<VectorStoreRecordVectorProperty>())
         {
             // Use index name same as vector property name with underscore
-            var indexName = $"{this._storagePropertyNames[property.PropertyName]}_";
+            var vectorPropertyName = this._storagePropertyNames[property.DataModelPropertyName];
```

Comment:
Don't these names have the BSON name as an override now?  I thought we said we weren't going to support that naming for now.  Not to be confused with my comment about the parsing / serialization where we probably do need to know about the BSON names just because we are using BSON parsing/serialization.  I realise all this is a bit confusing, so maybe we can just have a chat about it later.  It took me a while to get my head around this issue for JSON tbh.

Reply:
So, here we are using name of the vector property, and it should be exactly as in storage, because we need to associate index with that property. Based on latest discussion, it will be the same as name in `BsonElement` attribute, `StoragePropertyName` or `DataModelPropertyName`.